include/irr/asset/filters/dithering/CDither.h

```diff
@@ -0,0 +1,56 @@
+// Copyright (C) 2020 - AnastaZIuk
+// This file is part of the "IrrlichtBAW" engine.
+// For conditions of distribution and use, see copyright notice in irrlicht.h
+
+#ifndef __IRR_C_DITHER_H_INCLUDED__
+#define __IRR_C_DITHER_H_INCLUDED__
+
+#include "../include/irr/asset/IDither.h"
+
+namespace irr
+{
+	namespace asset
+	{
+		//! Base CRTP class for dithering classes
+		/*
+			There are several dithering classes:
+
+			- CWhiteNoiseDither
+			- CBayerMatrixDither
+			- CPrecomputedDither
+
+			Each of them put some noise on a processed image.
+		*/
+
+		template<class CRTP>
+		class CDither : public IDither
+		{
+			public:
+				CDither() {}
+				virtual ~CDither() {}
+
+				class CState : public IDither::IState
+				{
+					public:
+						CState() {}
+						virtual ~CState() {}
+
+						TexelRange texelRange;
```

Comment:
what is texel range?

Reply:
i will change it to extent maybe?

Reply:
oh ok so its a measure of dimension?

Reply:
It's extent + offset field

Reply:
ah kwl

Reply:
you only need extent to be honest... and also not sure what its doing in the CDither class